Building docs broken with pillow 5.4

With the new release of `pillow` the following error occurs:

```
generating gallery for gallery/images_contours_and_fields... [100%] watermark_image.py

Traceback (most recent call last):
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx/cmd/build.py", line 303, in build_main
    args.tags, args.verbosity, args.jobs, args.keep_going)
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx/application.py", line 263, in __init__
    self._init_builder()
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx/application.py", line 325, in _init_builder
    self.emit('builder-inited')
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx/application.py", line 510, in emit
    return self.events.emit(event, self, *args)
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx/events.py", line 80, in emit
    results.append(callback(*args))
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx_gallery/gen_gallery.py", line 221, in generate_gallery_rst
    seen_backrefs)
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx_gallery/gen_rst.py", line 462, in generate_dir_rst
    gallery_conf)
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx_gallery/gen_rst.py", line 711, in generate_file_rst
    save_thumbnail(image_path_template, src_file, file_conf, gallery_conf)
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx_gallery/gen_rst.py", line 425, in save_thumbnail
    scale_image(img, thumb_file, *gallery_conf["thumbnail_size"])
  File "/home/circleci/.local/lib/python3.6/site-packages/sphinx_gallery/gen_rst.py", line 386, in scale_image
    img = img.resize((width_sc, height_sc), Image.BICUBIC)
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/Image.py", line 1800, in resize
    im = self.convert(self.mode[:-1]+'a')
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/Image.py", line 915, in convert
    self.load()
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/ImageFile.py", line 250, in load
    self.load_end()
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/PngImagePlugin.py", line 677, in load_end
    self.png.call(cid, pos, length)
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/PngImagePlugin.py", line 140, in call
    return getattr(self, "chunk_" + cid.decode('ascii'))(pos, length)
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/PngImagePlugin.py", line 356, in chunk_IDAT
    raise EOFError
EOFError

Exception occurred:
  File "/home/circleci/.local/lib/python3.6/site-packages/PIL/PngImagePlugin.py", line 356, in chunk_IDAT
    raise EOFError
EOFError
The full traceback has been saved in /tmp/sphinx-err-f0iznxik.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
Makefile:30: recipe for target 'html' failed
make: *** [html] Error 2
Exited with code 2
```

It looks like some problem with saving a `.png` file somewhere along the way - maybe an issue with `sphinx_gallery`?

--- a/doc-requirements.txt
+++ b/doc-requirements.txt
@@ -11,6 +11,6 @@ colorspacious
 ipython
 ipywidgets
 numpydoc>=0.8
-pillow>=3.4,<5.4
+pillow>=3.4,!=5.4.0
 sphinx-gallery>=0.2
 sphinx-copybutton

